components/http_client.rst

```diff
@@ -417,6 +418,18 @@ When the HTTP status code of the response is in the 300-599 range (i.e. 3xx,
     // instead the original response content (even if it's an error message)
     $content = $response->getContent(false);
 
+
+Thus, there are two types of exceptions:
+
+* Exceptions implementing the ``HttpExceptionInterface`` are thrown when your code
+  does not handle the status codes in the 300-599 range.
+  When streaming responses, these exceptions do not interrupt the stream as you might
+  still want to read the content of these responses.
+  That being said, be aware that they are thrown on every single chunk.
```

Comment:
that's not accurate: chunks don't throw anything about status codes
but if you repeatidily call the same method (e.g. `$response->getHeaders()`) again and again, it's going to throw again and again for sure :)

Reply:
Yes, that's what I meant. So wording still needs to be improved :) Proposals?